src/kakarot/execution_context.cairo

```diff
@@ -38,7 +38,7 @@ namespace ExecutionContext {
         let initial_pc = 0;
         let gas_used = 0;
         // TODO: Add support for gas limit
-        let gas_limit = 0;
+        let gas_limit = 21002;
```

Comment:
Why do you hardcode the gas limit ? It should come from the user. We need to redefine a bit the execution context to add some data from the transaction. Please put on hold this PR for the moment

Reply:
Ok! I added the gas limit just to test the (Gas_limit - gas_used -2) , when it was gas_limit = 0 it was giving and overflow error

Reply:
Should i close it for now then?

Reply:
We can let it open. I will put a special status with a label

Reply:
So that you can resume the work once we design a solution. Thanks for your help

Reply:
Ok! Will make the changes later then :)